internals/overlord/planstate/manager.go

```diff
@@ -55,19 +56,41 @@ func NewManager(layersDir string) (*PlanManager, error) {
 // the case of a non-existent layers directory, or no layers in the layers
 // directory, an empty plan is announced to change subscribers.
 func (m *PlanManager) Load() error {
+	m.planLock.Lock()
+	if !reflect.DeepEqual(m.plan, &plan.Plan{}) {
```

Comment:
nit: This is a lot of code to effectively determine if we have ```Init``` or ```Load``` before, and it is also slightly odd in the sense that if someone explicitly init it to ```plan.Plan{}```, that is not recognized, and it will allow the ```Load``` to replace the explicitly set empty plan.

Would it not be simpler to just add a flag in the plan manager tracking this instead ? ```planInitialized bool``` ?

Reply:
```go
	plan, err := plan.ReadDir(m.layersDir)
	if err != nil {
		return err
	}

	m.planLock.Lock()
        if !m.planInitialized {
	    m.plan = plan
	    m.planInitialized = true            
	}
	m.planLock.Unlock()

	m.callChangeListeners(plan)
	return nil
```

Reply:
```
func (m *PlanManager) Load() (err error) {
    m.planLock.Lock()
    defer func() {
        m.planLock.Unlock()
        if err != nil {
            m.callChangeListeners(plan)
        }
    }
    
    plan, err := plan.ReadDir(m.layersDir)
    if err != nil {
	return err
    }
    if !m.planInitialized {
        m.plan = plan
        m.planInitialized = true
    }
    return nil
```